api/handler/v1beta1/grpc.go

```diff
@@ -172,7 +179,13 @@ func (s *GRPCServer) UpdateProvider(ctx context.Context, req *guardianv1beta1.Up
 		URN:    providerConfig.URN,
 		Config: providerConfig,
 	}
-	if err := s.providerService.Update(p); err != nil {
+
+	user, err := s.getUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	ctx = audit.WithActor(ctx, user)
```

Comment:
If this is done in every/most of API, should we move it to grpc interceptor?

Reply:
that sounds nice, let me check

Reply:
I created `audit.UnaryServerInterceptor` to avoid this repetition as suggested to use grpc interceptor